### ISSUE TITLE
Conflicting declarations of int64 and uint64 with OpenCV on Linux

### ISSUE THREAD

**Issue by lebarsfa** [SBG-Systems/sbgECom#6]
Hello,

When trying to use sbgEcom and OpenCV (popular computer vision library) in the same program, we get this error :
```
/home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgTypes.h:67:55: error: conflicting declaration ‘typedef long long unsigned int uint64’
   67 | SBG_DEPRECATED_TYPEDEF(typedef unsigned long long int uint64); // 64 bits
      |                                                       ^~~~~~
/home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgDefines.h:343:40: note: in definition of macro ‘SBG_DEPRECATED_TYPEDEF’
  343 |   #define SBG_DEPRECATED_TYPEDEF(decl) decl __attribute__((deprecated))
      |                                        ^~~~
In file included from /usr/include/opencv4/opencv2/core/cvdef.h:171,
                 from /usr/include/opencv4/opencv2/core.hpp:52,
                 from /usr/include/opencv4/opencv2/highgui.hpp:46,
                 from /home/user/Downloads/sbgECom-3.1.2358-stable/examples/ellipseMinimal/src/ellipseMinimal.cpp:36:
/usr/include/opencv4/opencv2/core/hal/interface.h:62:21: note: previous declaration as ‘typedef uint64_t uint64’
   62 |    typedef uint64_t uint64;
      |                     ^~~~~~
In file included from /home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgTypes.h:40,
                 from /home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgCommon.h:130,
                 from /home/user/Downloads/sbgECom-3.1.2358-stable/examples/ellipseMinimal/src/ellipseMinimal.cpp:39:
/home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgTypes.h:72:54: error: conflicting declaration ‘typedef long long int int64’
   72 | SBG_DEPRECATED_TYPEDEF(typedef signed long long int  int64);  // 64 bits
      |                                                      ^~~~~
/home/user/Downloads/sbgECom-3.1.2358-stable/cmake/../common/sbgDefines.h:343:40: note: in definition of macro ‘SBG_DEPRECATED_TYPEDEF’
  343 |   #define SBG_DEPRECATED_TYPEDEF(decl) decl __attribute__((deprecated))
      |                                        ^~~~
In file included from /usr/include/opencv4/opencv2/core/cvdef.h:171,
                 from /usr/include/opencv4/opencv2/core.hpp:52,
                 from /usr/include/opencv4/opencv2/highgui.hpp:46,
                 from /home/user/Downloads/sbgECom-3.1.2358-stable/examples/ellipseMinimal/src/ellipseMinimal.cpp:36:
/usr/include/opencv4/opencv2/core/hal/interface.h:61:20: note: previous declaration as ‘typedef int64_t int64’
   61 |    typedef int64_t int64;
      |                    ^~~~~
```

To reproduce on Ubuntu : 

- `sudo apt install libopencv-dev`
- Rename `ellipseMinimal.c` to `ellipseMinimal.cpp` and add in the beginning `#include "opencv2/highgui.hpp"`.
- Update `CMakeLists.txt` accordingly with : 
```cmake
find_package(OpenCV)
include_directories(${OpenCV_INCLUDE_DIRS})
[…]
add_executable(ellipseMinimal "${PROJECT_SOURCE_DIR}/../examples/ellipseMinimal/src/ellipseMinimal.cpp")
target_link_libraries(ellipseMinimal sbgECom ${OpenCV_LIBS})
```

Possible workaround in `sbgTypes.h` : 
```c++
#ifdef SBG_ENABLE_UINT64_DEPRECATED_TYPEDEF
SBG_DEPRECATED_TYPEDEF(typedef unsigned long long int	uint64);	// 64 bits
#endif // SBG_ENABLE_UINT64_DEPRECATED_TYPEDEF

SBG_DEPRECATED_TYPEDEF(typedef signed char				int8);		//  8 bits
SBG_DEPRECATED_TYPEDEF(typedef signed short				int16);		// 16 bits
SBG_DEPRECATED_TYPEDEF(typedef signed int				int32);		// 32 bits
#ifdef SBG_ENABLE_INT64_DEPRECATED_TYPEDEF
SBG_DEPRECATED_TYPEDEF(typedef signed long long int		int64);		// 64 bits
#endif // SBG_ENABLE_INT64_DEPRECATED_TYPEDEF
```
Or : 
```c++
//SBG_DEPRECATED_TYPEDEF(typedef unsigned long long int	uint64);	// 64 bits

SBG_DEPRECATED_TYPEDEF(typedef signed char				int8);		//  8 bits
SBG_DEPRECATED_TYPEDEF(typedef signed short				int16);		// 16 bits
SBG_DEPRECATED_TYPEDEF(typedef signed int				int32);		// 32 bits
//SBG_DEPRECATED_TYPEDEF(typedef signed long long int		int64);		// 64 bits

// Previous definitions were conflicting with some versions of OpenCV on Linux, 
// so the same definitions as OpenCV are copied here...
#if defined _MSC_VER || defined __BORLANDC__
SBG_DEPRECATED_TYPEDEF(typedef __int64 int64);
SBG_DEPRECATED_TYPEDEF(typedef unsigned __int64 uint64);
#else
#include <inttypes.h>
SBG_DEPRECATED_TYPEDEF(typedef int64_t int64);
SBG_DEPRECATED_TYPEDEF(typedef uint64_t uint64);
#endif
```

**Comment by rsiryani** [comment]
Thanks for the feedback, it will be fixed in next sbgECom 4.0 release.

**Comment by rsiryani** [comment]
Hello,
sbgECom 4.0 has been released and this issue should be fixed by now.
Thanks for the feedback.
